src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -248,7 +248,21 @@ private class TraceBinder : SerializationBinder
 
             public override System.Type BindToType(string assemblyName, string typeName)
             {
+                //TODO cache
                 var assemblies = AppDomain.CurrentDomain.GetAssemblies().Where(a => !a.IsDynamic);
+                var assemblyNameObj = new AssemblyName(assemblyName);
+                //find matching assemblies by name, version is not used.
+                var matchingAssembly = assemblies.FirstOrDefault(x => x.GetName().Name == assemblyNameObj.Name);
```

Comment:
You can use `assemblyName` here directly and save the need to create `assemblyNameObj` first?

Reply:
the latter - this is a way to remove the version and culture info from the string reliably.